src/pudl/extract/eia.py

```diff
@@ -0,0 +1,92 @@
+"""Run the PUDL ETL Pipeline.
+
+The PUDL project integrates several different public datasets into a well
+normalized relational database allowing easier access and interaction between all
+datasets. This module coordinates the extract/transfrom/load process for
+data from:
+
+ - US Energy Information Agency (EIA):
+   - Form 860 (eia860)
+   - Form 923 (eia923)
+ - US Federal Energy Regulatory Commission (FERC):
+   - Form 1 (ferc1)
+ - US Environmental Protection Agency (EPA):
+   - Continuous Emissions Monitory System (epacems)
+"""
+
+from dagster import AssetOut, Output, multi_asset
+
+import pudl
+from pudl.metadata.classes import DataSource
+
+logger = pudl.logging_helpers.get_logger(__name__)
+
+# TODO (bendnorman): Add this information to the metadata
+eia_raw_table_names = (
+    "raw_boiler_fuel_eia923",
+    "raw_boiler_generator_assn_eia860",
+    "raw_fuel_receipts_costs_eia923",
+    "raw_generation_fuel_eia923",
+    "raw_generator_eia860",
+    "raw_generator_eia923",
+    "raw_generator_existing_eia860",
+    "raw_generator_proposed_eia860",
+    "raw_generator_retired_eia860",
+    "raw_ownership_eia860",
+    "raw_plant_eia860",
+    "raw_stocks_eia923",
+    "raw_utility_eia860",
+)
+
+
+# TODO (bendnorman): Figure out type hint for context keyword and mutli_asset return
+@multi_asset(
+    outs={table_name: AssetOut() for table_name in sorted(eia_raw_table_names)},
```

Comment:
Yes :( one of the stranger requirements of dagster. I think it would be way simpler if the keys could be matched without needing to preserve the order.